src/config.js

```diff
@@ -312,7 +312,7 @@ export function newConfig() {
         return memo;
       }, {});
     }
-    return Object.assign({}, config);
+    return deepClone(config);
```

Comment:
This may not be backwards compatible, should we create either a param to allow to return Object.assign or a separate function so behaviour is explicit in this case?